server-protocol.md

```diff
@@ -92,6 +85,26 @@ following keys (and ignores all others):
   future version of the protocol requires a rate-limiting CAPTCHA ticket or
   other authorization record, the server can send `error` (explaining the
   requirement) if it does not see this ticket arrive before the `bind`.
+* `permission-required`: a set of available authentication methods,
+  proof of work challenges etc. The client needs to "solve" one of
+  them in order to get access to the service.
+
+The client should examine the `permissions-required` methods (if
+present) and select one to use (see also "Permission to Use the
+Server" below). If the client doesn't support any permissions methods
+or has already chosen not to use any then it _may_ send `bind`
```

Comment:
> or has already chosen not to use any

I see what you are going for, but somebody reading this may take this for a contradiction to the `none` permission type mentioned below (same thing for line 101). Let me try a different wording:

> The client should examine the `permissions-required` methods (if
present) and select one to use by replying with a `submit-permissions`
message (see also "Permission to Use the Server" below).
> 
> 1. If the client skips this step or the provided information is not satisfying,
  the server either grants usage permission regardless, or sends a 
  "permission denied" error message and close the connection.
> 2. If the client doesn't support any permissions methods, it either displays an
  error to the user directly or skips this step in the hope that the server is
  in a good mood (point 1.).
> 3. Backwards compatibility: Older clients will always send the `bind`
  message immediately (without even waiting for the `welcome` message).
  Server implementations must accept and treat this like point 1.

Reply:
okay that sounds better, I'll re-word further :)

Reply:
okay, I pushed 2a11d324a6a772ec56fd2bf98b9e40d0a2565324 with wording a lot like the above. thanks!